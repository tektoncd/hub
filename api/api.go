package apiserver

import (
	api "api/gen/api"
	"context"
	"log"
)

// api service example implementation.
// The example methods log the requests and return zero values.
type apisrvc struct {
	logger *log.Logger
}

// NewAPI returns the api service implementation.
func NewAPI(logger *log.Logger) api.Service {
	return &apisrvc{logger}
}

// Get all tasks and pipelines.
func (s *apisrvc) List(ctx context.Context) (err error) {
	s.logger.Print("api.list")
	return
}
