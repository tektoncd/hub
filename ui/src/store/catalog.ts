import { Instance, types } from 'mobx-state-tree';
import { Icons } from '../common/icons';

const icons: { [catalog: string]: Icons } = {
  official: Icons.Cat,
  verified: Icons.Certificate,
  community: Icons.User
};

export const Catalog = types
  .model({
    id: types.identifierNumber,
    name: types.optional(types.string, ''),
    type: types.optional(types.string, ''),
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }))
  .views((self) => ({
    get icon(): Icons {
      return icons[self.type] || Icons.Unknown;
    }
  }));

export type ICatalog = Instance<typeof Catalog>;
export type ICatalogStore = Instance<typeof CatalogStore>;

export const CatalogStore = types
  .model({
    items: types.map(Catalog)
  })

  .actions((self) => ({
    add(item: ICatalog) {
      self.items.put({ id: item.id, name: item.name, type: item.type });
    },
    clearSelected() {
      self.items.forEach((c) => {
        c.selected = false;
      });
    }
  }))

  .views((self) => ({
    get values() {
      return Array.from(self.items.values());
    },

    get selected() {
      const list = new Set();
      self.items.forEach((c: ICatalog) => {
        if (c.selected) {
          list.add(c.id);
        }
      });

      return list;
    }
  }));
