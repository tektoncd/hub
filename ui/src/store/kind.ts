import { types, Instance } from 'mobx-state-tree';
import { Icons } from '../common/icons';

const icons: { [kind: string]: Icons } = {
  Task: Icons.Build,
  Pipeline: Icons.Domain
};

export const Kind = types
  .model({
    name: types.identifier,
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }))
  .views((self) => ({
    get icon(): Icons {
      return icons[self.name] || Icons.Unknown;
    }
  }));

export type IKind = Instance<typeof Kind>;
export type IKindStore = Instance<typeof KindStore>;

export const KindStore = types
  .model({
    items: types.optional(types.map(Kind), {})
  })

  .actions((self) => ({
    add(item: string) {
      self.items.put({ name: item, selected: false });
    },

    clearSelected() {
      self.items.forEach((k) => {
        k.selected = false;
      });
    }
  }))

  .views((self) => ({
    get values() {
      return Array.from(self.items.values());
    },

    get selected() {
      const list = new Set();
      self.items.forEach((c: IKind) => {
        if (c.selected) {
          list.add(c.name);
        }
      });

      return list;
    }
  }));
