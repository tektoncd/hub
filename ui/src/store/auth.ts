import { flow, getEnv, Instance, types } from 'mobx-state-tree';
import { Api } from '../api';
import { titleCase } from '../common/titlecase';

export const TokenInfo = types.model('TokenInfo', {
  token: types.optional(types.string, ''),
  expiresAt: types.optional(types.number, 0),
  refreshInterval: types.optional(types.string, '')
});

export const Error = types.model('Error', {
  status: types.optional(types.number, 0),
  customMessage: types.optional(types.string, ''),
  serverMessage: types.optional(types.string, '')
});

export type ITokenInfo = Instance<typeof TokenInfo>;
export type IError = Instance<typeof Error>;

export interface AuthCodeProps {
  code: string;
}

interface Refresh {
  (): void;
}

export const AuthStore = types
  .model('AuthStore', {
    accessTokenInfo: types.optional(TokenInfo, {}),
    refreshTokenInfo: types.optional(TokenInfo, {}),
    isLoading: true,
    isAuthenticated: false,
    userRating: types.optional(types.number, 0),
    authErr: types.optional(Error, {}),
    ratingErr: types.optional(Error, {})
  })
  .actions((self) => ({
    addAccessTokenInfo(item: ITokenInfo) {
      self.accessTokenInfo.token = item.token;
      self.accessTokenInfo.expiresAt = item.expiresAt;
      self.accessTokenInfo.refreshInterval = item.refreshInterval;
    },
    addRefreshTokenInfo(item: ITokenInfo) {
      self.refreshTokenInfo.token = item.token;
      self.refreshTokenInfo.expiresAt = item.expiresAt;
      self.refreshTokenInfo.refreshInterval = item.refreshInterval;
    },
    setIsAuthenticated(l: boolean) {
      self.isAuthenticated = l;
    },
    setUserRating(rating: number) {
      self.userRating = rating;
    },
    setLoading(l: boolean) {
      self.isLoading = l;
    },
    setErrorMessage(error: IError) {
      switch (error.status) {
        case 400:
          error.customMessage = 'Bad Request';
          break;
        case 401:
          error.customMessage = 'Unauthorized. Please login again!';
          break;
        case 404:
          error.customMessage = 'Requested resource not found';
          break;
        case 500:
          error.customMessage = 'Internal server error. Please try again after some time!';
          break;
        default:
          error.customMessage = '';
      }
    },
    onFailure(err: Error) {
      self.authErr.serverMessage = err.toString();
      self.authErr.status = 400;
      self.authErr.customMessage = 'GitHub login failed';
    },
    logout() {
      localStorage.clear();
      self.isAuthenticated = false;
      self.isLoading = false;
      self.userRating = 0;
    }
  }))
  .views((self) => ({
    get api(): Api {
      return getEnv(self).api;
    }
  }))
  .actions((self) => ({
    authenticate: flow(function* (authCode: AuthCodeProps, refresh?: Refresh) {
      try {
        self.setLoading(true);

        const { api } = self;
        const json = yield api.authentication(authCode.code);

        const userDetails = json.data;
        self.addAccessTokenInfo(userDetails.access);
        self.addRefreshTokenInfo(userDetails.refresh);

        self.setIsAuthenticated(true);

        if (refresh) {
          refresh();
        }
      } catch (err) {
        const error: IError = {
          status: err.status,
          serverMessage: titleCase(err.data.message),
          customMessage: ''
        };
        self.setErrorMessage(error);
        self.authErr = error;
      }
      self.setLoading(false);
    }),

    getRating: flow(function* (resourceId: number) {
      try {
        self.setLoading(true);
        const { api } = self;

        if (self.isAuthenticated) {
          const json = yield api.getRating(resourceId, self.accessTokenInfo.token);
          self.setUserRating(json.rating);
        }
      } catch (err) {
        const error: IError = {
          status: err.status,
          serverMessage: titleCase(err.data.message),
          customMessage: ''
        };
        self.setErrorMessage(error);
        self.ratingErr = error;
      }
      self.setLoading(false);
    }),

    setRating: flow(function* (resourceId: number, rating: number) {
      try {
        self.setLoading(true);

        const { api } = self;

        if (self.isAuthenticated) {
          yield api.setRating(resourceId, self.accessTokenInfo.token, rating);
          self.setUserRating(rating);
        }
      } catch (err) {
        const error: IError = {
          status: err.status,
          serverMessage: titleCase(err.data.message),
          customMessage: ''
        };
        self.setErrorMessage(error);
        self.ratingErr = error;
      }
      self.setLoading(false);
    })
  }));
export type IAuthStore = Instance<typeof AuthStore>;
