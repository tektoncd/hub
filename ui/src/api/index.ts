import axios from 'axios';
import { API_URL } from '../config/constants';
import { ICategory } from '../store/category';
import { IResource, IVersion } from '../store/resource';

interface Token {
  token: string;
  refreshInterval: string;
  expiresAt: number;
}

interface TokenData {
  access: Token;
  refresh: Token;
}

export interface AuthResponse {
  data: TokenData;
}

export interface Rating {
  rating: number;
}

export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
  resourceVersion(resourceId: number): Promise<IVersion>;
  versionUpdate(versionId: number): Promise<IVersion>;
  authentication(authCode: string): Promise<AuthResponse>;
  readme(rawURL: string): Promise<string>;
  yaml(rawURL: string): Promise<string>;
  getRating(resourceId: number, token: string): Promise<Rating>;
  setRating(resourceId: number, token: string, rating: number): Promise<void | null>;
}

export class Hub implements Api {
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resources() {
    try {
      return axios.get(`${API_URL}/resources`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async authentication(authCode: string) {
    try {
      return axios
        .post(`${API_URL}/auth/login?code=${authCode}`)
        .then((response) => response.data)
        .catch((err) => Promise.reject(err.response));
    } catch (err) {
      return err;
    }
  }

  async resourceVersion(resourceId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/versions`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async versionUpdate(versionId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/version/${versionId}`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async readme(rawURL: string) {
    try {
      const URL = rawURL.substring(0, rawURL.lastIndexOf('/') + 1);
      return axios.get(`${URL}/README.md`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async yaml(rawURL: string) {
    try {
      const newLine = '\n';
      return axios.get(`${rawURL}`).then((response) => '```yaml' + newLine + response.data);
    } catch (err) {
      return err.response;
    }
  }

  async getRating(resourceId: number, token: string) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/rating`, {
          headers: {
            Authorization: `Bearer ${token}`
          }
        })
        .then((response) => response.data)
        .catch((err) => Promise.reject(err.response));
    } catch (err) {
      return err;
    }
  }

  async setRating(resourceId: number, token: string, rating: number) {
    try {
      return axios
        .put(
          `${API_URL}/resource/${resourceId}/rating`,
          { rating: rating },
          {
            headers: {
              Authorization: `Bearer ${token}`
            }
          }
        )
        .then((response) => response.data)
        .catch((err) => Promise.reject(err.response));
    } catch (err) {
      return err;
    }
  }
}
