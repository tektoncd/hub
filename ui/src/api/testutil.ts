import { Api } from './';
import * as fs from 'fs';
import { ICategory } from '../store/category';
import { IResource, IVersion } from '../store/resource';

export class FakeHub implements Api {
  dataDir: string;

  constructor(dataDir: string) {
    this.dataDir = dataDir;
  }

  async categories() {
    const data = `${this.dataDir}/categories.json`;
    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<ICategory>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }

  async resources() {
    const data = `${this.dataDir}/resources.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<IResource>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }

  async resourceVersion(resourceId: number) {
    const data = `${this.dataDir}/resource_versions_${resourceId}.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<IVersion>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }

  async versionUpdate(versionId: number) {
    const data = `${this.dataDir}/version_details_${versionId}.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<IVersion>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }
}
