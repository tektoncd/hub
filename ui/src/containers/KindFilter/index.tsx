import React from 'react';
import { observer } from 'mobx-react';
import Filter from '../../components/Filter';
import { useMst } from '../../store/root';

const KindFilter: React.FC = observer(() => {
  const { resources } = useMst();
  return <Filter store={resources.kinds} header="Kind" />;
});

export default KindFilter;
