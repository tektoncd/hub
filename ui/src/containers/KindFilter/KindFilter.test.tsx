import React from 'react';
import { mount } from 'enzyme';
import { FakeHub } from '../../api/testutil';
import { createProviderAndStore } from '../../store/root';
import { when } from 'mobx';
import { Checkbox } from '@patternfly/react-core';
import KindFilter from '.';

const TESTDATA_DIR = `src/store/testdata`;
const api = new FakeHub(TESTDATA_DIR);
const { Provider, root } = createProviderAndStore(api);

describe('KindFilter', () => {
  it('finds the filter component and matches the count', (done) => {
    const component = mount(
      <Provider>
        <KindFilter />
      </Provider>
    );

    const { resources } = root;
    when(
      () => {
        return !resources.isLoading;
      },
      () => {
        setTimeout(() => {
          const kinds = resources.kinds;
          expect(kinds.items.size).toBe(2);
          component.update();

          const kf = component.find(KindFilter);
          expect(kf.length).toEqual(1);

          expect(component.debug()).toMatchSnapshot();

          expect(component.find(Checkbox).length).toEqual(2);

          done();
        }, 0);
      }
    );
  });
});
