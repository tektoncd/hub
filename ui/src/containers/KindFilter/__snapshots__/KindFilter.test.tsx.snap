// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KindFilter finds the filter component and matches the count 1`] = `
"<Provider>
   
  <KindFilter>
    <Filter store={{...}} header=\\"Kind\\">
      <div>
        <Grid>
          <div className=\\"pf-l-grid\\" style={[undefined]}>
            <GridItem span={6}>
              <div className=\\"pf-l-grid__item pf-m-6-col\\" style={[undefined]}>
                <Text component=\\"h1\\" className=\\"hub-filter-header\\">
                  <h1 data-ouia-component-type=\\"PF4/Text\\" data-ouia-safe={true} data-ouia-component-id=\\"OUIA-Generated-Text-1\\" data-pf-content={true} className=\\"hub-filter-header\\">
                    Kind
                  </h1>
                </Text>
              </div>
            </GridItem>
            <GridItem span={3}>
              <div className=\\"pf-l-grid__item pf-m-3-col\\" style={[undefined]}>
                <Button variant=\\"plain\\" aria-label=\\"Clear\\" onClick={[Function: res] { _isMSTAction: true }}>
                  <ButtonBase innerRef={{...}} variant=\\"plain\\" aria-label=\\"Clear\\" onClick={[Function: res] { _isMSTAction: true }}>
                    <button onClick={[Function: res] { _isMSTAction: true }} aria-disabled={false} aria-label=\\"Clear\\" className=\\"pf-c-button pf-m-plain\\" disabled={false} tabIndex={[undefined]} type=\\"button\\" role={{...}} data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe={true} data-ouia-component-id=\\"OUIA-Generated-Button-plain-1\\">
                      <TimesIcon color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false}>
                        <svg style={{...}} fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 352 512\\" aria-labelledby={{...}} aria-hidden={true} role=\\"img\\">
                          <path d=\\"M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z\\" />
                        </svg>
                      </TimesIcon>
                    </button>
                  </ButtonBase>
                </Button>
              </div>
            </GridItem>
            <GridItem span={12}>
              <div className=\\"pf-l-grid__item pf-m-12-col\\" style={[undefined]}>
                <Checkbox className=\\"hub-filter-checkbox\\" label={{...}} isChecked={false} onChange={[Function: onChange]} aria-label=\\"controlled checkbox\\" id=\\"Task\\" name=\\"Task\\" isValid={true} isDisabled={false} ouiaSafe={true}>
                  <div className=\\"pf-c-check hub-filter-checkbox\\">
                    <input id=\\"Task\\" name=\\"Task\\" className=\\"pf-c-check__input\\" type=\\"checkbox\\" onChange={[Function (anonymous)]} aria-invalid={false} aria-label=\\"controlled checkbox\\" disabled={false} checked={false} data-ouia-component-type=\\"PF4/Checkbox\\" data-ouia-safe={true} data-ouia-component-id=\\"OUIA-Generated-Checkbox-1\\" />
                    <label className=\\"pf-c-check__label\\" htmlFor=\\"Task\\">
                      <div>
                        <Icon id=\\"build\\" size=\\"sm\\" label=\\"Task\\">
                          <BuildIcon size=\\"sm\\" className=\\"hub-icon\\" label=\\"Task\\" color=\\"currentColor\\" noVerticalAlign={false}>
                            <svg style={{...}} fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 1024 1024\\" aria-labelledby={{...}} aria-hidden={true} role=\\"img\\" className=\\"hub-icon\\" label=\\"Task\\">
                              <path d=\\"M996.377115,171.480135 L540.537115,5.18013521 C531.409589,1.76249044 521.743498,0.00809818406 511.997115,6.72397778e-05 C501.871295,-0.0125134433 491.820803,1.74040061 482.297115,5.18013521 L27.6671155,171.370135 C11.0249457,177.434467 -0.0238641555,193.287629 0.046775475,211.000135 L0.046775475,249.000135 C0.195841208,267.102495 11.623122,283.189328 28.6671155,289.290135 L85.2071155,309.500135 C95.0094893,313.010977 105.907101,311.533188 114.420477,305.53861 C122.933853,299.544031 127.998417,289.782269 127.997115,279.370135 L127.997115,234.580135 L511.997115,95.0001352 L895.177115,235.170135 L895.177115,295.370135 C895.178887,302.102977 898.453489,308.414744 903.95698,312.293243 C909.460471,316.171742 916.505874,317.132839 922.847115,314.870135 L995.477115,289.270135 C1012.59546,283.170258 1024.01857,266.9528 1023.99715,248.780135 L1023.99715,210.870135 C1023.97475,193.258354 1012.96683,177.531031 996.427115,171.480135 M533.647115,346.370135 L511.997115,353.370135 L489.587115,348.080135 C479.853994,344.894147 469.183221,346.571799 460.897291,352.59073 C452.611361,358.609661 447.71662,368.238859 447.737051,378.480135 L447.737051,417.580135 C447.725709,431.081528 456.204913,443.131767 468.917115,447.680135 L499.577115,458.180135 C507.177328,460.919505 515.534404,460.669293 522.957115,457.480135 L554.327115,445.870135 C567.027983,441.306886 575.500688,429.265885 575.507257,415.770135 L575.507257,376.770135 C575.537565,366.523106 570.643619,356.885253 562.352331,350.863869 C554.061043,344.842484 543.381903,343.17069 533.647115,346.370135 M351.737115,359.870135 L351.737115,318.770135 C351.726919,304.92777 342.817563,292.661266 329.657115,288.370135 L265.717115,267.480135 C255.985877,264.299431 245.319589,265.977891 237.035494,271.993509 C228.751399,278.009128 223.854266,287.632278 223.86709,297.870135 L223.86709,337.000135 C223.874371,350.47567 232.322855,362.502666 244.997115,367.080135 L308.937115,390.080135 C318.749212,393.583471 329.652547,392.090624 338.161859,386.078784 C346.67117,380.066943 351.720981,370.28887 351.697115,359.870135 M991.217115,442.000135 L856.317115,394.650135 C852.839791,393.339817 849.00444,393.339817 845.527115,394.650135 L798.077115,411.550135 C783.887115,416.550135 783.887115,436.650135 798.077115,441.650135 L913.157115,483.000135 L511.997115,646.780135 L109.477115,482.780135 L223.867115,441.780135 C237.947115,436.690135 237.947115,416.690135 223.867115,411.690135 L176.607115,394.780135 C173.112329,393.58999 169.321902,393.58999 165.827115,394.780135 L33.1571155,442.370135 C13.0440455,449.54441 -0.269196877,468.728185 0.0471154616,490.080135 L0.0471154616,503.080135 C-0.847277689,523.712017 11.059679,542.763147 29.9971155,551.000135 L476.407115,747.280135 C498.111167,756.808317 522.807966,756.840942 544.537115,747.370135 L802.537115,634.470135 L994.287115,549.930135 C1013.05026,541.718274 1024.80919,522.789452 1023.85712,502.330135 L1023.85712,489.430135 C1024.36395,468.22962 1011.19974,449.100375 991.217115,442.000135 M995.527115,672.000135 L915.657115,643.810135 C911.820884,642.433761 907.604002,642.573251 903.867115,644.200135 L861.307115,662.810135 C848.027115,668.610135 848.727115,687.610135 862.307115,692.500135 L910.997115,710.000135 L511.997115,916.170135 L111.277115,709.870135 L152.437115,695.080135 C165.827115,690.280135 166.717115,671.580135 153.837115,665.580135 L112.837115,646.370135 C109.008929,644.580285 104.630262,644.368356 100.647115,645.780135 L27.9971155,671.870135 C11.3473237,677.858232 0.19240487,693.586867 0.0471154616,711.280135 L0.0471154616,730.080135 C0.0632949619,745.406045 8.42616142,759.506405 21.8671155,766.870135 L470.707115,1013.47014 C483.376418,1020.36989 497.570829,1023.98983 511.997115,1024.00014 C526.111827,1023.98457 539.988079,1020.35974 552.307115,1013.47014 L1002.19712,766.870135 C1015.63034,759.501145 1023.98494,745.401803 1023.99712,730.080135 L1023.99712,712.470135 C1023.99712,694.324323 1012.60558,678.131732 995.527115,672.000135 M799.277301,336.480135 L799.277301,297.480135 C799.312025,287.231444 794.419143,277.590499 786.126499,271.568129 C777.833854,265.54576 767.152103,263.875935 757.417115,267.080135 L693.487115,288.000135 C680.330167,292.290121 671.421636,304.551456 671.407115,318.390135 L671.407115,359.390135 C671.416959,369.788712 676.478869,379.534105 684.980399,385.521966 C693.481929,391.509827 704.362857,392.993413 714.157115,389.500135 L778.097115,366.700135 C790.819287,362.10717 799.292115,350.02599 799.277301,336.500135\\" />
                            </svg>
                          </BuildIcon>
                        </Icon>
                        Task
                      </div>
                    </label>
                  </div>
                </Checkbox>
                <Checkbox className=\\"hub-filter-checkbox\\" label={{...}} isChecked={false} onChange={[Function: onChange]} aria-label=\\"controlled checkbox\\" id=\\"Pipeline\\" name=\\"Pipeline\\" isValid={true} isDisabled={false} ouiaSafe={true}>
                  <div className=\\"pf-c-check hub-filter-checkbox\\">
                    <input id=\\"Pipeline\\" name=\\"Pipeline\\" className=\\"pf-c-check__input\\" type=\\"checkbox\\" onChange={[Function (anonymous)]} aria-invalid={false} aria-label=\\"controlled checkbox\\" disabled={false} checked={false} data-ouia-component-type=\\"PF4/Checkbox\\" data-ouia-safe={true} data-ouia-component-id=\\"OUIA-Generated-Checkbox-2\\" />
                    <label className=\\"pf-c-check__label\\" htmlFor=\\"Pipeline\\">
                      <div>
                        <Icon id=\\"domain\\" size=\\"sm\\" label=\\"Pipeline\\">
                          <DomainIcon size=\\"sm\\" className=\\"hub-icon\\" label=\\"Pipeline\\" color=\\"currentColor\\" noVerticalAlign={false}>
                            <svg style={{...}} fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 1024 1024\\" aria-labelledby={{...}} aria-hidden={true} role=\\"img\\" className=\\"hub-icon\\" label=\\"Pipeline\\">
                              <path d=\\"M544,511.9 C561.476067,511.948945 575.664266,525.962873 575.994253,543.371439 L576,543.9 L576,991.9 C575.951055,1009.37607 561.937127,1023.56427 544.528561,1023.89425 L544,1023.9 L352,1023.9 C334.523933,1023.85105 320.335734,1009.83713 320.005747,992.428561 L320,991.9 L320,543.9 C320.048945,526.423933 334.062873,512.235734 351.471439,511.905747 L352,511.9 L544,511.9 Z M704,576 L704,960 L640,960 L640,576 L704,576 Z M512,896 L384,896 L384,960 L512,960 L512,896 Z M960,513.1 L960,513.2 C960,665.4 884.1,799.9 768,880.9 L768,880.9 L768,799.4 C773.3,794.6 778.4,789.8 783.5,784.7 C855.624209,712.728348 896.108133,614.991161 896,513.1 L896,513.1 L960,513.1 Z M149.8,385 C100.714385,524.561202 135.973879,679.905126 240.5,784.6 C245.5,789.7 250.7,794.6 256,799.3 L256,799.3 L256,880.8 C139.9,799.8 63.9999013,665.3 63.9999013,513.1 C63.9712238,469.725105 70.236499,426.575548 82.6,385 L82.6,385 Z M512.1,704 L384,704 L384,832 L512.1,832 L512.1,704 Z M512,576 L448,576 L448,640 L512,640 L512,576 Z M864,-1.13686838e-13 C881.496381,-1.13686838e-13 895.713128,14.0417829 895.995713,31.4708215 L896,32 L896,406 C896,423.496381 881.958217,437.713128 864.529179,437.995713 L864,438 L608,438 C590.503619,438 576.286872,423.958217 576.004287,406.529179 L576,406 L576,32 C576,14.5036191 590.041783,0.286872383 607.470821,0.00428708574 L608,-1.13686838e-13 L864,-1.13686838e-13 Z M1024,64 L1024,384 L960,384 L960,64 L1024,64 Z M832,256 L640,256 L640,352 L832,352 L832,256 Z M224,-5.68434189e-14 C241.476067,0.0489452394 255.664266,14.0628726 255.994253,31.4714395 L256,32 L256,288 C255.951055,305.476067 241.937127,319.664266 224.528561,319.994253 L224,320 L32,320 C14.5239331,319.951055 0.335734448,305.937127 0.00574685353,288.528561 L0,288 L0,32 C0.0489452394,14.5239331 14.0628726,0.335734448 31.4714395,0.00574685353 L32,-5.68434189e-14 L224,-5.68434189e-14 Z M192,192 L64,192 L64,256 L192,256 L192,192 Z M512,65 L512,129 C460.642759,128.90985 409.795141,139.181409 362.5,159.2 C347.917219,165.399771 333.724426,172.47947 320,180.4 L320,180.4 L320,108.2 C380.012207,79.7623165 445.590913,65.0071076 512,65 L512,65 Z M832,64 L736,64 L736,160 L832,160 L832,64 Z M192,64 L128,64 L128,128 L192,128 L192,64 Z\\" />
                            </svg>
                          </DomainIcon>
                        </Icon>
                        Pipeline
                      </div>
                    </label>
                  </div>
                </Checkbox>
              </div>
            </GridItem>
          </div>
        </Grid>
      </div>
    </Filter>
  </KindFilter>
   
</Provider>"
`;
